### ISSUE TITLE
`upgrade`: `Error: Download failed: An unexpected error occurred while moving files to the final destination.`

### ISSUE THREAD

**Issue by brokosz** [mas-cli/mas#483]
## Your Environment

- `mas version`: 1.8.6
- macOS version (`system_profiler SPSoftwareDataType -detailLevel mini`): 13.0.1 (22A400)

### mas Install Method

- [x] `brew install mas` (homebrew-core)
- [] [mas-cli/tap](https://github.com/mas-cli/homebrew-tap)
- [] `.pkg` installer from [releases](https://github.com/mas-cli/mas/releases)
- [] Built from source
  - Fork/branch: ? (e.g. mas-cli/main)
  - Xcode version: 10.?

## Describe the Bug

> Error: Download failed: An unexpected error occurred while moving files to the final destination.

Consistently getting the above error when running `mas upgrade`. Started post-Ventura upgrade on my m1 mbp. 

## To Reproduce

1. Go to `Terminal`
1. Run `mas upgrade`
1. See error

## Screenshots, Terminal Output

> If applicable, add screenshots to help explain your problem.

```bash
$ mas upgrade
Upgrading 3 outdated applications:
TestFlight (3.2.2), Flighty (2.5.1), Prime Video (9.22)
==> Downloading TestFlight
==> Downloading Flighty – Fast Flight Tracker
==> Installed Flighty – Fast Flight Tracker
==> Downloading Amazon Prime Video
==> Installed Amazon Prime Video
Error: Download failed: An unexpected error occurred while moving files to the final destination.
```

## Additional Context

It's strange that several apps will (re)install properly, but the final app on the list will get this. Doesn't matter if it's one app or four, whichever app is last will throw the above error.

**Comment by donileo** [comment]
Any update on how to fix this issue? Getting the same thing.

**Comment by YDA93** [comment]
I have the bug with Testflight as well

**Comment by scottpeterson** [comment]
I have the bug with Testflight as well

**Comment by mr-pinzhang** [comment]
got same issue for `Testflight` app.

**Comment by rgoldberg** [comment]
@brokosz @isadon @YDA93 @scottpeterson @mr-pinzhang Is anyone still experiencing this issue?

**Comment by brokosz** [comment]
Yes on a different device with the same mas and os version let me know what data I can get you. 

----- Original message -----

**Comment by rgoldberg** [comment]
@brokosz Thanks for helping.

You said that you had the problem for the last app in the list, but others mentioned that they had the problem specifically for TestFlight.

TestFlight was the first app in your list, not the last. The error was output after the other apps had successfully been installed.

Do you remember what led you to think it was the last app, not specifically TestFlight?

To investigate:

Start up the Mac or restart it if it is already on (trying to avoid transient fixes/issues for mas from possible prior mas usage in the same login session).

Don't run any mas commands other than those specified below before you've finished running the listed commands.

What is the output of the following commands? If everything upgrades fine for any `mas upgrade`, then you can skip subsequent `mas upgrade` commands.

Before you do this, if there are any additional steps you think should be done before, or interspersed with, these commands, please let me know. After you've run all the commands below, if you think that running any other mas commands might provide useful info, then you can go ahead and run them (some commands seem to resolve issues that other commands experience, so I'm trying to avoid the (unlikely) situation where a command fixes the problem before the outlined investigation is complete).

```
# verify mas version
mas version

# verify macOS version
system_profiler SPSoftwareDataType -detailLevel mini

# check permissions
ls -dlh /Applications/*.app

# list installed apps
mas list

# attempt to upgrade all apps
mas upgrade

# attempt to upgrade whatever apps failed in previous upgrade of all apps
mas upgrade

# list outdated apps
mas outdated

# attempt to upgrade apps after running mas outdated (that at times makes a difference)
mas upgrade

# list outdated apps again
mas outdated

# for each <app ID> from the prior outdated list, attempt to install it by itself (e.g., <app ID> = 899247664 for TestFlight)
mas upgrade <app ID>

# check permissions
ls -dlh /Applications/*.app
```

**Comment by brokosz** [comment]
Oh wait sorry - so I'm having the JSON parsing error still, not this one. Sorry for the confusion. I've got two working systems now which is good! 

On Sep 14, 2024, at 10:28, Ross Goldberg ***@***.***> wrote:

DuckDuckGo removed one tracker. More <https://duckduckgo.com/-hGysyZFupsqvAi0nzhKiHW7Lo9C3OqfyFK-9EKafjgjwJ94lbiiYYWOepPP9rUWE_FmHRtwFtOhrYBjZAnOochywx0cSHAbK2w0-tQWQrLUYY2UoOhpWamSxqE6ZF-yFe9BRoCCu0ksW5VnE6UHvHGY_-8nJMf4VbV2YQA1DY0-IENZgVa6DfDiAmzGJ3-8Dyc9J52V-Y0NfOp_rH65j0Y-1DSuMjigkq-GtlLY89hau4I0xRXh_wLelaWf4BrRlDXKfMxV2lkZdrw-igFS0CWCyZsgDYqF8q63HN2bN5CWLJbl7i07bBqwEL6pFkIDyj3gBtaAM7NEJ8ZMVh11A21TruQRMYDuhVJKHkyQG7MSblI8JPJvpiQbavBVAz9UKWsjaVhqpKZCV2pFMm-OAMg1zsP82oiZsPEpc5BBK878w0CWWkHadyG6G4_JQaPTzXGxoDsciM_PIz0gkbd9tRLLFhaZYtoOwHvkbnC0na68yUpA94CKZ8n419FBjcXk4sQvr3ryBJw9J8R_g-B9tWQ4uvSlyFECIebdAB> 
Deactivate  <https://duckduckgo.com/> 

@brokosz <https://github.com/brokosz> Thanks for helping.

You said that you had the problem for the last app in the list, but others mentioned that they had the problem specifically for TestFlight.

TestFlight was the first app in your list, not the last. The error was output after the other apps had successfully been installed.

Do you remember what led you to think it was the last app, not specifically TestFlight?

To investigate:

Start up the Mac or restart it if it is already on (trying to avoid transient fixes/issues for mas from possible prior mas usage in the same login session).

Don't run any mas commands other than those specified below before you've finished running the listed commands.

What is the output of the following commands? If everything upgrades fine for any mas upgrade, then you can skip subsequent mas upgrade commands.

Before you do this, if there are any additional steps you think should be done before, or interspersed with, these commands, please let me know. After you've run all the commands below, if you think that running any other mas commands might provide useful info, then you can go ahead and run them (some commands seem to resolve issues that other commands experience, so I'm trying to avoid the (unlikely) situation where a command fixes the problem before the outlined investigation is complete).

# verify mas version
mas version

# verify macOS version
system_profiler SPSoftwareDataType -detailLevel mini

# check permissions
ls -dlh /Applications/*.app

# list installed apps
mas list

# attempt to upgrade all apps
mas upgrade

# attempt to upgrade whatever apps failed in previous upgrade of all apps
mas upgrade

# list outdated apps
mas outdated

# attempt to upgrade apps after running mas outdated (that at times makes a difference)
mas upgrade

# list outdated apps again
mas outdated

# for each <app ID> from the prior outdated list, attempt to install it by itself (e.g., <app ID> = 899247664 for TestFlight)
mas upgrade <app ID>

# check permissions
ls -dlh /Applications/*.app

**Comment by rgoldberg** [comment]
@brokosz Thanks for the info.

Do you remember why you thought it was the last listed app, instead of TestFlight itself?

Knowing about that will help us solve this longterm.

If you ever experience this issue again, please let us know.

Hopefully, other reporters will reply, too.

**Comment by rgoldberg** [comment]
@Azd325  @brokosz @isadon @mr-pinzhang @scottpeterson @YDA93

I was just able to upgrade to the new TestFlight 3.6.0 without any problems using mas 1.8.6 from Homebrew.

I ran `mas upgrade 899247664` instead of `mas upgrade`.

Can you run `mas upgrade` with mas 1.8.6?

Did any of you experience problems upgrading TestFlight to 3.6.0?

Thanks.

**Comment by rgoldberg** [comment]
@Azd325  @brokosz @isadon @mr-pinzhang @scottpeterson @YDA93

Does mas 1.8.7 fix this issue?

**Comment by brokosz** [comment]
@rgoldberg 1.8.7 is working well for me so far 👍🏼

**Comment by rgoldberg** [comment]
Closing as I haven't heard back from anyone. If you can reproduce the problem with the latest version of mas, please reopen this issue. Thanks.
